src/Text/Parser/Token.hs

```diff
@@ -706,7 +701,7 @@ integer' :: TokenParsing m => m Integer
 integer' = int <?> "integer"
 {-# INLINE integer' #-}
 
-sign :: TokenParsing m => m (Integer -> Integer)
+sign :: (TokenParsing m, Num a) => m (a -> a)
```

Comment:
I understand the intent behind this, but it is a somewhat unforced change, as it's not required to fix #83. Moreover, the existing convention in this module is to use `Integer` everywhere, so I'd favor sticking to that convention.